apps/api/migrations/expire-at/expire-at-delay.migration.spec.ts

```diff
@@ -0,0 +1,147 @@
+  let query;
+
+  beforeEach(async () => {
+    session = new UserSession();
+    await session.initialize();
+    digestTemplate = await createDigestTemplate(session);
+    delayTemplate = await createDelayTemplate(session);
+    query = {
+      _environmentId: session.environment._id,
+      _organizationId: session.organization._id,
+      expireAt: { $exists: false },
+    };
+
+    for (let i = 0; i < 5; i++) {
+      const newSubscriberIdInAppNotification = SubscriberRepository.createObjectId();
+      await sendTrigger(session, digestTemplate, newSubscriberIdInAppNotification);
+      await sendTrigger(session, delayTemplate, newSubscriberIdInAppNotification);
+      await new Promise((r) => setTimeout(r, 1000));
+      await sendTrigger(session, digestTemplate, newSubscriberIdInAppNotification);
+    }
+    await new Promise((r) => setTimeout(r, 1000));
+
+    await messageRepository.update({ _environmentId: session.environment._id }, { $unset: { expireAt: 1 } });
+    await notificationRepository.update({ _environmentId: session.environment._id }, { $unset: { expireAt: 1 } });
+    await jobRepository.update({ _environmentId: session.environment._id }, { $unset: { expireAt: 1 } });
+    await executionDetailsRepository.update({ _environmentId: session.environment._id }, { $unset: { expireAt: 1 } });
+  });
+
+  it('should not add expireAt for a pending execution', async function () {
+    await notificationExpireAt(query);
+
+    const notifications = await notificationRepository.find({
+      _environmentId: session.environment._id,
+      _templateId: delayTemplate?._id,
+    });
+    const jobs = await jobRepository.find({ _environmentId: session.environment._id, _templateId: delayTemplate?._id });
+    const executionDetails = await executionDetailsRepository.find({
+      _environmentId: session.environment._id,
+      _notificationTemplateId: delayTemplate?._id,
+    });
+
+    notifications.forEach((msg) => {
+      expect(msg.expireAt).to.not.exist;
+    });
+    jobs.forEach((msg) => {
+      expect(msg.expireAt).to.not.exist;
+    });
+    executionDetails.forEach((msg) => {
+      expect(msg.expireAt).to.not.exist;
+    });
+  });
+
+  it('should add expireAt to pending events that were digested', async function () {
+    await session.awaitRunningJobs(digestTemplate?._id, false, 5);
+
+    await notificationExpireAt(query);
+
+    const notifications = await notificationRepository.find({
+      _environmentId: session.environment._id,
+      _templateId: digestTemplate?._id,
+    });
+    const jobs = await jobRepository.find({
+      _environmentId: session.environment._id,
+      _templateId: digestTemplate?._id,
+    });
+    const executionDetails = await executionDetailsRepository.find({
+      _environmentId: session.environment._id,
+      _notificationTemplateId: digestTemplate?._id,
+    });
+
+    notifications.forEach((msg) => {
+      expect(msg.expireAt).to.exist;
```

Comment:
should we also check if the date value is what we expect it to be?

Reply:
Sure, I can add that